### PR TITLE
Log requests

### PR DESCRIPTION
**Pull request by iarspider**
This is super fragile (it relies on exact signature of `Github.__init__`) and ugly (because Python **at runtime** automatically mangles names of class members that start with double underscore by prefixing them with actual class name)

### CODE REVIEW

File: process_pr.py [hunk 1]
```diff
@@ -995,7 +997,7 @@ def process_pr(repo_config, gh, repo, issue, dryRun, cmsbuild_user=None, force=F
         watchers = set([gh_user_char + u for u in watchers])
         print("Watchers " + ", ".join(watchers))
 
-        all_commits = get_pr_commits_reversed(pr)
+        all_commits = [x for x in pr.get_commits()][::-1]
```
Comment (smuzaffar) [inline process_pr.py:1000]: @iarspider , can you please create a separate PR for this change.

File: process_pr.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import sys
```
Comment (smuzaffar) [inline process_pr.py:1]: ??

File: process-pull-request.py [hunk 3]
```diff
@@ -66,16 +155,18 @@
...
         import repo_config
 
         if not getattr(repo_config, "RUN_DEFAULT_CMS_BOT", True):
             sys.exit(0)
-        gh = Github(login_or_token=get_gh_token(opts.repository), per_page=100)
+        if opts.debug:
+            gh = DebugGithub(login_or_token=get_gh_token(opts.repository), per_page=100)
+        else:
+            gh = Github(login_or_token=get_gh_token(opts.repository), per_page=100)
+
```
Comment (smuzaffar) [inline process-pull-request.py:169]: this change looks bit complex . Is there no other way to achieve this?

Comment (iarspider) [reply process-pull-request.py:169]: Didn't think of it, but yes, it should work.

**Comment by iarspider** [comment]
hold
need to test latest changes, and maybe also only log requests

File: github_utils.py [hunk 4]
```diff
@@ -1003,3 +1003,28 @@ def get_pr_commits_reversed(pr):
         except IndexError:
             print("Index error: May be PR with no commits")
     return []
+
+
+def enable_github_loggin():
+    import logging
+
+    class MyHandler(logging.Handler):
+        level = 0
```
Comment (smuzaffar) [inline github_utils.py:1012]: @iarspider , I think you can set it to `logging.DEBUG` to old get the calls for debug messages

**Comment by smuzaffar** [comment]
@iarspider , can we call logging debug from our `github_api` function

**Comment by iarspider** [comment]
> @iarspider , can we call logging debug from our `github_api` function

This is one of the things I plan doing at some point - use logging module at least in `process_pr` (and `github_api`), but maybe everywhere

**Comment by smuzaffar** [comment]
> > @iarspider , can we call logging debug from our `github_api` function
> 
> This is one of the things I plan doing at some point - use logging module at least in `process_pr` (and `github_api`), but maybe everywhere

fir github_api function it should be as trivial as calling `logging.debug(msg)`

**Comment by iarspider** [comment]
unhold

**Comment by smuzaffar** [comment]
looks good, can you please update the PR description

**Comment by smuzaffar** [comment]
@iarspider , I am merging it. Can you please update the Pr description?

**Pull request merged**